go/libraries/doltcore/env/actions/infer_schema.go

```diff
@@ -181,7 +182,19 @@ func leastPermissiveType(strVal string, floatThreshold float64) typeinfo.TypeInf
 		return typeinfo.BoolType
 	}
 
-	return typeinfo.StringDefaultType
+	if strings.Contains(strVal, "{") {
```

Comment:
JSON data could also come in as a JSON array, instead of a JSON map, for example:

```sql
mysql> create table json1 (pk int primary key, j1 json);
Query OK, 0 rows affected (0.00 sec)

mysql> insert into json1 values (1, "[1,2,3]");
Query OK, 1 row affected (0.01 sec)
```

It would be good to look for `[` as another hint that a column's data may be in JSON format and attempt a test parse, too.